packages/airnode-abi/src/decoding.test.ts

```diff
@@ -0,0 +1,69 @@
+import { ethers } from 'ethers';
+import * as decoding from './decoding';
+
+describe('decode', () => {
+  it('decodes address types', () => {
+    const data =
+      '0x316100000000000000000000000000000000000000000000000000000000000054657374416464726573734e616d6500000000000000000000000000000000000000000000000000000000004128922394c63a204dd98ea6fbd887780b78bb7d';
+    const res = decoding.decode(data);
+    expect(res).toEqual({ TestAddressName: '0x4128922394C63A204Dd98ea6fbd887780b78bb7d' });
+  });
+
+  it('decodes int256 types', () => {
+    const data =
+      '0x316900000000000000000000000000000000000000000000000000000000000054657374496e744e616d65000000000000000000000000000000000000000000fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffc18';
+    const res = decoding.decode(data);
+    expect(res).toEqual({ TestIntName: ethers.BigNumber.from('-1000') });
+  });
+
+  it('decodes uint256 types', () => {
+    const data =
+      '0x31750000000000000000000000000000000000000000000000000000000000005465737455496e744e616d65000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000007d0';
+    const res = decoding.decode(data);
+    expect(res).toEqual({ TestUIntName: ethers.BigNumber.from('2000') });
+  });
+
+  it('decodes bytes types', () => {
+    const data =
+      '0x31420000000000000000000000000000000000000000000000000000000000005465737442797465734e616d650000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000020536f6d652062797465732076616c756500000000000000000000000000000000';
+    const res = decoding.decode(data);
+    expect(res).toEqual({ TestBytesName: 'Some bytes value' });
```

Comment:
A correct `bytes` example would be a hexadecimal string (and the number of hexadecimals has to be even)